app/controllers/chargeback_controller.rb

```diff
@@ -255,8 +255,19 @@ def cb_rate_form_field_changed
     cb_rate_get_form_vars
     render :update do |page|                    # Use JS to update the display
       changed = (@edit[:new] != @edit[:current])
+      # Update the new column with the code of the currency selected by the user
+      @edit[:new][:details].each_with_index do |_detail, i|
+        next unless @edit[:new][:details][i][:currency] != @edit[:current][:details][i][:currency]
+        @edit[:current][:details][i][:currency] = @edit[:new][:details][i][:currency]
+        params[:id_column] = i
+        id_currency = @edit[:current][:details][i][:currency]
+        currency = ChargebackRateDetailCurrency.find_by(:id => id_currency)
+        params[:code_currency] = currency.code
+        page.replace("column_currency_#{i}", :partial => "cb_new_currency_column")
```

Comment:
@jrafanie Yes, you're right. It is clearer in this way. 
@edit[:new][:details] has all the rate_details so as the currency is the same for all the rate_details, I have selected the first one before the loop in order to fix it.